kbase-extension/static/kbase/js/widgets/function_output/kbaseReportView.js

```diff
@@ -442,206 +377,140 @@ define([
+                li = html.tag('li'),
+                a = html.tag('a'),
+                div = html.tag('div'),
+                br = html.tag('br', { close: false })
+
+            const linkList = ul({},
+                htmlLinks.map((link) => {
+                    const linkText = link.label || link.name;
+                    return li({}, [
+                        a({
+                            href: link.url,
+                            target: '_blank',
+                            ariaLabel: `open ${linkText} in another window`
+                        },
+                            linkText,
+                        ),
+                        link.description ? (br() + link.description) : ''
+                    ])
+                })
+            );
+            return div({}, linkList);
+        },
+
+        /**
+         * Returns a promise that resolves when each internal component completes rendering.
+         */
         render: function () {
-            const self = this;
             const _this = this;
             const t = html.tag,
                 div = t('div'),
                 a = t('a');
-            const ui = UI.make({ node: self.$mainPanel.get(0) });
-            const report = self.reportData;
+            const ui = UI.make({ node: this.$mainPanel.get(0) });
+            const report = this.reportData;
             const events = Events.make({
-                node: self.$mainPanel.get(0),
+                node: this.$mainPanel.get(0),
             });
+            const renderPromises = [];
 
-            // Handle warnings?
-            if (report.warnings) {
-                if (report.warnings.length > 0) {
-                    const $warningPanel = $(
-                        '<div style="max-height:100px;overflow-y:auto;margin:0px 5px 5px 10px;">'
-                    );
-                    const warnings = report.warnings;
-                    if (warnings.length >= 5) {
-                        $warningPanel.append(
-                            $('<div>')
-                                .css('margin', '5px')
-                                .append('[' + warnings.length + 'Warnings]')
-                        );
-                    }
-                    for (let k = 0; k < warnings.length; k++) {
-                        $warningPanel.append(
-                            $('<div>')
-                                .css('margin', '0px 5px 5px 10px')
-                                .append(
-                                    $('<span>').addClass('label label-warning').append(warnings[k])
-                                )
-                        );
-                    }
-                    self.$mainPanel.append($warningPanel);
-                }
+            // Handle warnings
+            if (report.warnings && report.warnings.length) {
+                this.$mainPanel.append(this.buildReportWarnings(report.warnings));
             }
 
-            if (self.options.showCreatedObjects) {
```

Comment:
This is replaced by the newer `cellComponents/tabs/results/outputWidget` component.